lib.go

```diff
@@ -113,6 +113,15 @@ func (vm *VM) GetMetrics() (*types.Metrics, error) {
 	return api.GetMetrics(vm.cache)
 }
 
+func validateGasForDeserialization(gasLimit uint64, deserCost types.UFraction, gasUsed *uint64, data []byte) (uint64, error) {
+	gasForDeserialization := deserCost.Mul(uint64(len(data))).Floor()
+	if gasLimit < gasForDeserialization+*gasUsed {
+		return *gasUsed, fmt.Errorf("Insufficient gas left to deserialize contract execution result (%d bytes)", len(data))
+	}
+	*gasUsed += gasForDeserialization
+	return *gasUsed, nil
```

Comment:
Since it passed `gasUsed` to the function as an `address(*uint64)`, it doesn't need to include it in the return. Why not just return an error or modify gasUsed to take only `value(uint64)`?

Reply:
Thank you. As you said, it's an unnecessary return value. It has been deleted.